pkg/charts/additionalchart.go

```diff
@@ -82,6 +87,11 @@ func (c *AdditionalChart) RevertMainChanges(pkgFs billy.Filesystem) error {
 	if err != nil {
 		return fmt.Errorf("encountered error while trying to get the main chart's working directory: %s", err)
 	}
+	if c.CRDChartOptions.UseTarArchive {
+		if err := filesystem.UnarchiveTgz(pkgFs, filepath.Join(c.WorkingDir, path.ChartExtraFileDir, path.ChartCRDTgzFilename), "", filepath.Join(c.WorkingDir, c.CRDChartOptions.CRDDirectory), false); err != nil {
+			return fmt.Errorf("encountered error while trying to unarchive CRD files from %s: %s", filepath.Join(c.WorkingDir, path.ChartExtraFileDir, path.ChartCRDTgzFilename), err)
+		}
+	}
 	// copy CRD files from packages/<package>/charts-crd/crd-manifest/ back to packages/<package>/charts/crds/
 	if err := helm.CopyCRDsFromChart(pkgFs, c.WorkingDir, c.CRDChartOptions.CRDDirectory, mainChartWorkingDir, path.ChartCRDDir); err != nil {
```

Comment:
I took another look at this. `CopyCRDsFromChart` calls `filesystem.CopyDir`, which if I understand correctly does not throw an error if the source directory does not exist. However I find this surprising, and I think most people would - typically, copy operations return an error if the source file/directory does not exist.

I think we need to address this - if `UseTarArchive` is `true`, then `ApplyMainChanges` will have deleted the CRD directory in the CRD chart, which means this call would return an error if `CopyCRDsFromChart` operated as expected.

What do you think?

Reply:
I took another look at this. I don't think that this will return an error, which is unintuitive - I've never seen a copy operation that doesn't return an error when the source file/directory does not exist. But to the best of my understanding, that is how `helm.CopyCRDsFromChart` works.

I'd prefer to check `UseTarArchive` before calling `helm.CopyCRDsFromChart`, but I'll leave it up to you.

Reply:
Yes its because we are leveraging `filepath.WalkdiDir` which seems like it doesn't err if the given dir doesn't exist. I'll just add a check in `helm.COpyCRDsFromChart` so we are covering uses in other charts as well.